packages/core/src/provider/base.ts

```diff
@@ -444,7 +444,7 @@ export class SmartAccountProvider<
       );
     }
 
-    request.signature = (await this.account.signMessage(
+    request.signature = (await this.account.signUserOperationHash(
```

Comment:
out of curiosity, did this need to change?

Reply:
yea this is the main usecase of this change. we don't want to couple `this.signMessage` to signing user ops